modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateProcessor.java

```diff
@@ -121,7 +122,16 @@ public DateProcessor create(Map<String, Processor.Factory> registry, String proc
             Locale locale = Locale.ENGLISH;
             if (localeString != null) {
                 try {
-                    locale = (new Locale.Builder()).setLanguageTag(localeString).build();
```

Comment:
This does not have the desired result. This snippet evaluates as false.

`LocaleUtils.parse("FRENCH").getLanguage().equals(Locale.FRENCH.getLanguage())`

The locale from LocaleUtils has the language "french" while Locale.FRENCH has the language "fr". This seems to be the key difference that results in the JodaTime parser not correctly looking for words in the correct language.

Perhaps it makes sense to move the logic that I have written here to convert from the display name for a language to the ISO code.